main.js

```diff
@@ -0,0 +1,112 @@
+const daysOfWeek = ['Sunday', 'Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday'];
+const currentDate = new Date();
+
+document.getElementById('submit-search').addEventListener('click', function () {
+    cityName = document.getElementById('location').value;
+    fetchFromCityLocation(cityName);
+});
+
+const fetchWeatherData = function (lat, lon, city) {
+    const fetchUrl = `https://api.openweathermap.org/data/2.5/forecast?lat=${lat}&lon=${lon}&appid=ca90637f0ed0bcfecc469a57ea89c230`;
+
+    fetch(fetchUrl, {
+        method: 'GET',
+        dataType: 'json'
+    }).then(data => data.json()).then(data => readWeatherData(data, city));

```

Comment:
Data is too generic, consider using something specific like weatherData
```suggestion
    }).then(weatherData => weatherData.json()).then(jsonWeatherData => readWeatherData(jsonWeatherData, city));

```